### PR TITLE
doc: add a notice when deploying k8s v1.24

### PR DESCRIPTION
**Pull request by 24sama**
### What type of PR is this?

/kind documentation

### What this PR does / why we need it:
 add a notice when deploying k8s v1.24

### Which issue(s) this PR fixes:

#1274 

### Special notes for reviewers:
```
```

### Does this PR introduced a user-facing change?

```release-note

```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/bootstrap/confirm/tasks.go [hunk 1]
```diff
@@ -236,8 +248,22 @@ Warning:
 	}
 	fmt.Println()
 
-	reader := bufio.NewReader(os.Stdin)
+	if k8sVersion, err := versionutil.ParseGeneric(u.KubeConf.Cluster.Kubernetes.Version); err == nil {
+		if cri, ok := u.PipelineCache.GetMustString(common.ClusterNodeCRIRuntimes); ok {
+			k8sV124 := versionutil.MustParseSemantic("v1.24.0")
+			if k8sVersion.AtLeast(k8sV124) && versionutil.MustParseSemantic(currentK8sVersion).LessThan(k8sV124) && strings.Contains(cri, "docker") {
+				fmt.Println("[Notice]")
+				fmt.Println("Dockershim has been removed from the Kubernetes project as of release 1.24.")
```
Comment (Patrick-LuoYu) [inline pkg/bootstrap/confirm/tasks.go:256]: Kubernetes v1.24 and later no longer support dockershim.

File: pkg/bootstrap/confirm/tasks.go [hunk 2]
```diff
@@ -236,8 +248,22 @@ Warning:
...
+				fmt.Println("Please make sure you have completed the migration from dockershim to other container runtimes that are compatible with the Kubernetes CRI.")
```
Comment (Patrick-LuoYu) [inline pkg/bootstrap/confirm/tasks.go:257]: Please make sure you have completed the migration from Docker to other container runtimes that are compatible with the Kubernetes CRI.

File: pkg/bootstrap/confirm/tasks.go (hunk 1 above)
Comment (24sama) [reply pkg/bootstrap/confirm/tasks.go]: Done.

File: pkg/bootstrap/confirm/tasks.go (hunk 2 above)
Comment (24sama) [reply pkg/bootstrap/confirm/tasks.go]: Done.

**Pull request merged**